lib/possible_bipartition.rb

```diff
@@ -1,4 +1,67 @@
 
+# Time Complexity - O(v + e) Using BFS, the function checks each vertex through the queue once, and all edges 
+# associated with that queue once or twice at most. Checking for non biparition-able graphs via edges beforehand
+# and only adding to the queue any edges with vertices that haven't been visited as a vertex prevents
+# repeat vertices from being visited. 
+
+# Space Complexity - O(v) - A hash, and set, and a queue are created where the max number of elements added will never exceed
+# the number of vertices (v) in the graph. group_hash will always have two keys between which the vertices are divided into based on 
+# edges, to_visit is a set of all vertices to delete from (in O(1) time), while bfs at worst will have all vertices in the Queue,
+# as the function doesn't add any edges that have already been visited. to_visit mostly exists to track any vertices 
+# in a graph that aren't reached by a group of other vertices in the graph, which group_hash and bfs_q can't account for. 
 def possible_bipartition(dislikes)
-  raise NotImplementedError, "possible_bipartition isn't implemented yet"
+  return true if dislikes.empty?
+  
+  # at this point will have at least one vertex
+  group_hash = {1 => Set.new, 2 => Set.new}
+  to_visit = Set.new(0...dislikes.length)
+  bfs_q = Queue.new
+
+  # set current group for leading vertex
+  v_group = 1
+
+  # set current group for edges
+  e_group = 2
```

Comment:
This works to label the groups, I just used the symbols `:red`, and `:green` to label the groups.